python/mage/node_classification/utils/extract_from_database.py

```diff
@@ -0,0 +1,258 @@
+import torch
+from torch_geometric.data import Data, HeteroData
+import mgp
+import numpy as np
+from tqdm import tqdm
+import random
+from collections import Counter
+import torch_geometric.transforms as T
+import typing
+from collections import defaultdict
+
+
+def nodes_fetching(
+    ctx: mgp.ProcCtx, features_name: str, class_name: str, data: HeteroData
+) -> typing.Tuple[HeteroData, typing.Dict, typing.Dict, str]:
+    """
+    This procedure fetches the nodes from the database and returns them in HeteroData object.
+
+    Args:
+        ctx: The context of the procedure.
+        features_name: The name of the features field.
+        class_name: The name of the class field.
+        data: HeteroData object where nodes will be stored.
+
+    Returns:
+        Tuple of (HeteroData, reindexing, inv_reindexing, observed_attribute).
+    """
+
+    nodes = list(iter(ctx.graph.vertices))  # obtain nodes from context
+    node_types = []  # variable for storing node types
+    embedding_lengths = defaultdict(int)  # variable for storing embedding lengths
```

Comment:
Shouldn't embedding lengths be same for all the nodes?

Reply:
I left user it has option to have different size features for different types.